notebooks/python/neo4j/README.md

```diff
@@ -0,0 +1,253 @@
+
+## Pandas and DataFrames
+At the outset, let’s provide some orientation about tabular data structures in the modern data science stack.  Traditional python programs execute on a single machine and all result-sets must fit in memory unless they are streamed to a collection of files, message broker, database, or other store.  The standard way to pass rowsets from one module to another is via an in-memory table called a Panda, or confusingly, Pandas DataFrame.  Common visualization libraries such as Seaborn, consume pandas and the Neo4j GDS and Python (with coercion) client libraries, return them.
+
+Spark, however, is a distributed, parallel processing computing platform.  Jobs can be executed on a single machine by specifying ``partition=1``, but parallelized data processing requires a distributed dataset which is called simply a (Spark) DataFrame, which is immutable.   If a (Spark) DataFrame can fit into memory on one node, it can be converted to a Panda.  The reverse is always true.  The Neo4j Spark Connector returns (Spark) DataFrames from queries.
+
+## Comparing APIs
+The examples exercise several pythonic APIs.  Which one should you use?  As shown in the truth table below, the GDS client is the obvious choice for GDS projects that don’t use Spark.  
+
+![API selector truth table](images/neo4j_dataproc_api_selector.png)
+
+The GDS client API allows executing GDS function calls natively as well as any Cypher statement using ``run_cypher``.  Calling GDS natively function uses slightly different object syntax than writing out Cypher strings using the gds client.
+
+When working distributed data on Spark cluster, the Neo4j Spark Connector provides read and write support like other Spark Connectors.  Although the Spark Connector supports GDS calls using the Cypher ``CALL`` syntax, the best practice is to instantiate the GDS or python client as well to handle script calls more elegantly.
+
+## Using the GDS client
+The GDS api is Neo4j's newest, most elegant and compact python API.  Usage requires importing a single library and instantiating an object on one line.
+
+    from graphdatascience import GraphDataScience
+
+    neo4j_url = "bolt://<server>:7687"
+    neo4j_user = "<user>"
+    neo4j_password = "<password>"
+
+    gds = GraphDataScience(neo4j_url, auth=(neo4j_user, neo4j_password))
+
+Thereafter, executing Cypher statements of any kind simply requires calling:
+
+    gds.run_cypher('<Cypher statement>')
+
+For GDS functions, invocation is fluent such as:
+
+    g_msa, result = gds.graph.project(
+        'msa-graph',
+        'MSA',
+        '*',
+        nodeProperties = [
+            "logPopulation",
+            "logMedianHouseholdIncome",
+            "logMedianHomePrice",
+            "logPercentOver25WithBachelors"])
+
+In this case, `g_msa` is a handle to the in-memory projection.
+
+Writing data back from the in-memory graph is simple:
+
+    gds.graph.writeRelationship(
```

Comment:
Please surround with code block, trailing back-slashes are needed for multi-line statement